herd/memUtils.ml

```diff
@@ -366,6 +366,10 @@ let lift_proc_info i evts =
   let not_speculated es e = not (E.EventSet.mem e es.E.speculated)
   let collect_reg_loads es = collect_by_loc es E.is_reg_load_any
   and collect_reg_stores es = collect_by_loc es E.is_reg_store_any
+  and collect_gpreg_and_spsysreg_loads es = collect_by_loc es (fun e ->
+    E.is_non_sysreg_load_any e || E.is_spsysreg_load_any e)
```

Comment:
I think it would be more clear to define this as
```
and collect_gpreg_and_spsysreg_loads es = collect_by_loc es (fun e ->
    (E.is_load_any e) && (E.is_spsysreg e || (not (E.is_sysreg e)))
```
This should inform what functions you need from E and which ones you should remove.